docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -20,196 +45,193 @@ The scenarios of defining placement rules in SQL include:
 
-They’re all achieved by executing `ALTER TABLE` statements.
+## Detailed Design
 
-### Add placement rules
+### New Syntax Overview
 
-Adding new replicas can be done by one or more `ADD PLACEMENT POLICY` clauses:
+There are two ways to specify placement rules:
 
-```sql
-ALTER TABLE table_name
-	ADD PLACEMENT POLICY CONSTRAINTS=constraints ROLE=role REPLICAS=replicas,
-	...
-```
+1. By assigning `PLACEMENT` directly on a database, table or partition (direct assignment)
+2. By creating a new `PLACEMENT POLICY` and then applying the placement policy to a database, table or partition (placement policy)
 
-This statement indicates TiDB to add replicas for all data of table `table_name`, including indexes.
+Using a `PLACEMENT POLICY` will be recommended for compliance requirements, since it can allow administrators to better keep track of usage. This can be seen as similar to how complex environments will use `ROLES` for management instead of directly assigning privileges to users.
 
-`ADD PLACEMENT POLICY` is just a part of alter options, just like `ADD COLUMN` or `ADD CONSTRAINT`.
+Both syntaxes are considered a [`table_option`](https://dev.mysql.com/doc/refman/8.0/en/alter-table.html), and available in both `CREATE TABLE` and `ALTER TABLE` contexts.
 
-To define multiple roles at once, multiple `ADD PLACEMENT POLICY` clauses can appear in a single `ALTER TABLE` statement, even for the same Raft role. For example:
+#### Direct assignment
+
+Creating a new table with a directly assigned `PLACEMENT` constraints:
 
 ```sql
-ALTER TABLE table_name
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=leader REPLICAS=1,
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=sh]" ROLE=follower REPLICAS=1
-	ADD PLACEMENT POLICY CONSTRAINTS="[+zone=gz]" ROLE=follower REPLICAS=1;
+CREATE TABLE t1 (
+	id INT NOT NULL PRIMARY KEY,
+	b VARCHAR(100)
+) PLACEMENT CONSTRAINTS="[+zone=sh]" ROLE=leader REPLICAS=1, CONSTRAINTS="[+zone=sh]" ROLE=follower REPLICAS=1, CONSTRAINTS="[+zone=gz]" ROLE=follower REPLICAS=1;
 ```
 
-This statement indicates PD to schedule the leader to `sh`, add one follower to  `sh` and one to `gz`. Note that as the leader can be only one, the first clause doesn't actually add a replica, so this statement adds 2 replicas.
-
-`ADD PLACEMENT POLICY` also supports adding TiFlash replicas for a table, as statement `ALTER TABLE table_name SET TIFLASH REPLICA count` does. For example:
+Adding `PLACEMENT` to an existing table:
 
 ```sql
-ALTER TABLE table_name
-	ADD PLACEMENT POLICY CONSTRAINTS="[+engine=tiflash]" ROLE=learner REPLICAS=1;
+CREATE TABLE t1 (
+	id INT NOT NULL PRIMARY KEY,
+	b VARCHAR(100)
+);
+ALTER TABLE t1 PLACEMENT CONSTRAINTS="[+zone=sh]" ROLE=leader REPLICAS=1, CONSTRAINTS="[+zone=sh]" ROLE=follower REPLICAS=1, CONSTRAINTS="[+zone=gz]" ROLE=follower REPLICAS=1;
```

Comment:
Some of the examples have multiple `ROLE=`, `REPLICAS=` and `CONSTRAINTS=` options
for which the order seems to be important. What about replacing this with a JSON
structure? That can more easily be parsed and changed by applications.

Also these names seem to be very generic, so it could happen (but is unlikely)
that MySQL would use the same in a next version.

Reply:
we can actually make them a json structure into CONSTRANTS, like

```
ALTER TABLE t1 PLACEMENT CONSTRAINTS+="{'zone': '+sh', 'role': 'leader', 'replicas': 1}",  CONSTRAINTS+="{'zone': '+sh', 'role': 'follower', 'replicas': 1}", CONSTRAINTS+="{'zone': '+gz', 'role': 'follower', 'replicas': 1}";

Reply:
in this case, CONSTRANTS are actually a list of JSON objects.

Reply:
futher modification can happen directly on the CONSTRANTS object.

Reply:
This has been updated with new proposed syntax, PTAL again thx :-)